Automattic/woocommerce-services
# Clicking "Maybe later" will dismiss feature announcement for 3 days

## Description
Clicking "Maybe later" will dismiss the modal for 3 days. 
![image](https://github.com/Automattic/woocommerce-services/assets/572862/6e4c15aa-b6be-4873-8079-11a3e9132f58)

### Technical decisions
If the user clicks "x" or "Maybe later", I consider this as "not eligible for migration". Thus, I added a cookie check in `is_eligible_for_migration()` as well. 

### Related issue(s)
N/A

### Steps to test
1. Pull this branch, go to `woocommerce-services/classes/class-wc-connect-service-settings-store.php`, change `is_eligible_for_migration()` to
```
		public function is_eligible_for_migration() {
			WC_Connect_Options::delete_option( 'wcshipping_migration_state' );
			$migration_state = WC_Connect_Options::get_option( 'wcshipping_migration_state' );
...
		}
```
The idea is to delete the `wcshipping_migration_state` option every time so we can re-test infinitely. 
2. Go to an order page and click "Create new label"
3. Observe the feature announcement modal
![image](https://github.com/Automattic/woocommerce-services/assets/572862/6f6f119d-33bf-410b-9c06-844b9ad4fd52)
4. Click "Maybe later", you should see the cookie `wcst-wcshipping-migration-dismissed` being set.
![image](https://github.com/Automattic/woocommerce-services/assets/572862/776d7e26-f739-4ff2-9d5e-d4a7c2ef941a)
5. Close the shipping modal, click "Create new label" again, and confirm no feature announcement migration modal.
6. Refresh the page (ctrl-r), click "Create new label", and confirm no feature announcement migration modal.
7. Mimic the 3-days cookie expiration by deleting the cookie `wcst-wcshipping-migration-dismissed`, refresh the page, and confirm feature announcemnt migration modal pop up again.

### Checklist
